Build failure: phpExtensions.opentelemetry 1.0.2 on MacOS Sonoma

### Steps To Reproduce

Steps to reproduce the behavior:
1. build opentelemetry extension on MacOS
```
nix build github:r-ryantm/nixpkgs/a399db43527b481d2af275ba058a4dc1d36ed373#php81Extensions.opentelemetry
```

### Build log

```
error: builder for '/nix/store/gsma784mx1iy2sbnbvd2wdxkld1asc3z-php-opentelemetry-1.0.2.drv' failed with exit code 2;
       last 10 log lines:
       >              ~~~~~~~~~~^~~~~~~~~~~
       > /private/tmp/nix-build-php-opentelemetry-1.0.2.drv-1/source/ext/otel_observer.c:231:24: note: remove extraneous parentheses around the comparison to silence this warning
       >         if ((type_mask == IS_UNDEF)) {
       >             ~          ^          ~
       > /private/tmp/nix-build-php-opentelemetry-1.0.2.drv-1/source/ext/otel_observer.c:231:24: note: use '=' to turn this equality comparison into an assignment
       >         if ((type_mask == IS_UNDEF)) {
       >                        ^~
       >                        =
       > 1 error generated.
       > make: *** [Makefile:211: otel_observer.lo] Error 1
       For full logs, run 'nix-store -l /nix/store/gsma784mx1iy2sbnbvd2wdxkld1asc3z-php-opentelemetry-1.0.2.drv'.
```

### Additional context

This occurred after this PR was merged: https://github.com/NixOS/nixpkgs/pull/303437

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@drupol 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.4.0, macOS 14.4.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/Users/avraham.appel/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/php-packages/opentelemetry/default.nix
+++ b/pkgs/development/php-packages/opentelemetry/default.nix
@@ -15,6 +15,8 @@ in buildPecl rec {
 
   sourceRoot = "${src.name}/ext";
 
+  env.NIX_CFLAGS_COMPILE = "-Wno-parentheses-equality";
+
   doCheck = true;
 
   meta = with lib; {

